refactor(oxfmt/lsp): move lsp code to `apps/oxfmt`, remove lsp formatter feature flag

Sysix opened the pull request:
> This PR refactors the LSP formatter code by moving it from crates/oxc_language_server to apps/oxfmt, removing the formatter feature flag from the language server crate.

This is now possible, because `oxc_language_server` is not anymore shipped https://github.com/oxc-project/oxc/pull/17457

Sysix commented:
* **#17462** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/17462?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/17462?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=17462).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Dec 30, 2:17 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 30, 2:17 AM UTC**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Sysix committed:
refactor(oxfmt/lsp): move lsp code to `apps/oxfmt`, remove lsp formatter feature flag (#17462)

> This PR refactors the LSP formatter code by moving it from crates/oxc_language_server to apps/oxfmt, removing the formatter feature flag from the language server crate.

This is now possible, because `oxc_language_server` is not anymore shipped https://github.com/oxc-project/oxc/pull/17457
